### PR TITLE
Add NXP Kinetis Pulse Width Timer (PWM capture) support

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
Add PWM capture driver for the NXP Kinetis Pulse Width Timer (PWT).

It depends on the following PRs:

- [x] https://github.com/zephyrproject-rtos/hal_nxp/pull/73
- [x] https://github.com/zephyrproject-rtos/zephyr/pull/31148
- [x] https://github.com/zephyrproject-rtos/zephyr/pull/26025

### CODE REVIEW

**Review by mbolivar-nordic — APPROVED** [review]
DT related bits seem reasonable to me.

File: west.yml [hunk 1]
```diff
@@ -102,7 +102,7 @@ manifest:
       revision: 41132e9220f8bc1223084975350c5e5f3b492afe
       path: tools/net-tools
     - name: hal_nxp
-      revision: 61a7788ac9379bfa7711bf9d15de735230528ea5
+      revision: pull/73/head
```
Comment (MaureenHelm) [inline west.yml:105]: The hal PR is now merged, please update this to the new SHA

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Looks good, just need to update the manifest

File: west.yml (hunk 1 above)
Comment (henrikbrixandersen) [reply west.yml]: Thanks @MaureenHelm. Updated!

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**